mesonbuild/dependencies/mpi.py

```diff
@@ -195,6 +217,33 @@ def _sanitize_version(self, out: str) -> str:
         return out
 
 
+class MSMPIConfigToolDependency(_MPIConfigToolDependency):
+    """Wrapper around MSYS2 MS-MPI mpicc et al"""
+
+    version_arg = '-show'  # dummy just to pass through factory without error
+
+    def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any],
+                 language: T.Optional[str] = None):
+        super().__init__(name, env, kwargs, language=language)
+        if not self.is_found:
+            return
+
+        args = self.get_config_value(['-show'], 'link and compile args')
+        self.compile_args = self._filter_compile_args(args)
+        self.link_args = self._filter_link_args(args)
+
+    def _sanitize_version(self, out: str) -> str:
+        # disregard dummy "out" input as MinGW + MS-MPI doesn't work like other platforms
+        if not shutil.which('mpiexec'):
```

Comment:
This seems... odd. The whole point of the ConfigToolDependency is to do this kind of stuff safely, and with machine file support. This really makes me think that using `ConfigToolDependency` as the parent is the right thing to do.

Reply:
OK I didn't refresh my understanding after the major refactoring you did some months ago. That would be better than abusing the factory function this way.